src/BuddyManager.cpp

```diff
@@ -651,6 +661,20 @@ void BuddyManager::emailSend(CNSocket* sock, CNPacketData* data) {
 
     // handle items
     std::vector<sItemBase> attachments;
+
+    if (pkt->iCash || pkt->aItem[0].ItemInven.iID) {
```

Comment:
I'm not positive, but I'm pretty sure you can put an attachment in any of the other slots and thus, the index will not necessarily be 0. It's a little sloppy, but checking all the slots in the condition is an acceptable solution:
```suggestion
    if (pkt->iCash || pkt->aItem[0].ItemInven.iID || pkt->aItem[1].ItemInven.iID || pkt->aItem[2].ItemInven.iID || pkt->aItem[3].ItemInven.iID) {
```

Reply:
Checking in-game, it does not seem to be possible to place items into a slot other than the first one available, which would make the index always 0. Keeping this line as is would make it cleaner. Your thoughts?

Reply:
Turns out you're right. All good with keeping it clean. Nice work.